packages/forms/src/components/Repeater/Repeater.tsx

```diff
@@ -89,12 +97,37 @@ const Fieldset = ({
   onRemove,
   children,
   disabled,
+  numOfLockedItems,
 }: RepeaterFieldsetProps) => {
   const intl = useIntl();
   const shouldReduceMotion = useReducedMotion();
   const { announce } = useAnnouncer();
+
+  const {
+    formState: { errors, isDirty },
+  } = useFormContext();
+
+  const fieldErrors = errors[name] as FieldErrors<{
+    [key: string]: FieldError;
+  }>;
+  const hasError = useMemo(() => {
+    if (fieldErrors && isDirty) {
+      return (
+        Object.keys(fieldErrors).includes(`${index}`) && fieldErrors[index]
+      );
+    }
+    return false;
+  }, [fieldErrors, index, isDirty]);
+
   // Non-zero index position of the fieldset for humans
   const position = index + 1;
+  const isFirstItem = disabled || index <= 0;
+  const isLastItem = disabled || index === total - 1;
+
+  const isLocked = numOfLockedItems ? index + 1 <= numOfLockedItems : false;
+  const previousItemLocked = numOfLockedItems
+    ? numOfLockedItems + 1 === index + 1
+    : false;
```

Comment:
Well I'm assuming were only locking things that take priority which would usually mean they would start at the top. Also my logic was you can't lock on item on an index 3 if the other 2 don't exist, and if there are two items above the 3rd locked item then they would probably be locked as well. However, I'm happy to hear your thoughts on it.

Reply:
Okay, I guess maybe I don't really understand the locking part 😕 . If it has to be the first X items would we not just make that not part of the repeater? 🤔 My understanding would be a user could select to lock the position of any item. Take the following scenario,

User has a list of 5 items, they want to ensure one is always in position 3 so they "lock" it. They then move item 2 down which would make it skip over 3 and put it in position 4 and move position 4 up to 2.

Reply:
@substrae Seems as though we are having trouble understanding exactly how the lock feature should be working. Could you help us out?

Reply:
@yonikid15 @esizer 

Ahh, sorry, this was a miscommunication on our part. The locking feature was implemented as a part of Jerbo's assessment record work, so it probably hasn't been fully explained. The intent here is for the system to be able to lock items, not the user. The reason this feature exists is because the locked items are still cognitively part of the list, but their order is uniquely constrained due to external factors. Locked items will, to my knowledge, always be at the top of the list, but they might not be added in that order.

For example, in Jerbo's design, you have a repeater of assessment types. The first type is **always** "Application", so that guy is locked in the first spot on page load. The user can then proceed to add other assessment types. If, however, they choose to add "Screening questions at the time of application", that assessment gets added and locked into the second spot, regardless of which place in the order they added it it.

After speaking with Jerbo, we don't foresee any need to lock items at specific indices, though it might be helpful down the road to reverse this and be able to lock items to the bottom of the list.

Reply:
So this is part of the AC for the [form/card repeater](https://github.com/GCTC-NTGC/gc-digital-talent/issues/8161) and not the fieldset repeater? Or do both need the locking feature?

Reply:
Are the not the same component with different children passed to them 🤔

Reply:
No, they need to be separate because one represents repeating forms while the other one is repeating fieldsets so we can't re-use the same component 👎

Reply:
So the locking feature is only necessary for both components or just the CardRepeater? From the designs I've seen the locking feature only seems to be present in the soon to be built CardRepeater.

Reply:
Okay, I'm a bit confused, or maybe I'm misunderstanding. The presentation layer and the functionality of the repeater component are generally indifferent to the context they exist in:
- look and feel of the list, cards, errors, add item button, etc.
- the functionality of moving an item up or down
- the functionality of locking an item in place
- the functionality to set a maximum or minimum number of items
- the functionality to accept child content for each card

The child content is passed when the repeater is imported into the UI, and how that content is handled (submission, etc.) should be handled by contextual variants, whether through props or other means.

If I was going to build this, it would look something like:
- Repeater
  - RepeaterCoreAndStyles
  - RepeaterDataHandling
    - FieldsetVariant
    - StaticVariant

To me this is more scaleable and reduces repetition by centralizing the majority of the interaction and UI while providing extensible options based on the context. Unless the intent is to break up the styles/core functionality and have your two unique components import them? Even then, that seems like it would be hard to follow for new devs.

Reply:
The issue is, one is a form (or at least appears to be) and one is not (currently it is a `fieldset`) so it is impossible to make them the same component since the underlying HTML and functionality would be completly different even though it appears to be similar in the design. 

We are leveraging [`useFieldArray`](https://react-hook-form.com/docs/usefieldarray) from `react-hook-form` to handle things like inserting, removing and moving items. It is required to be used inside of a form. Since the new use case is not a repeatable `fieldset` being used in a form and instead a form itself, we can't use this component and the best we can do is share a few of the styles between them.

We need will need to create a component to repeat forms that emit an `onSave` event with the ability to toggle between a display and edit mode and act as it's own independent item.

Unless you are suggesting that the new changes are not forms, cannot be saved individually and instead are saved by a single submit handler.

Reply:
Hmmm... maybe @Jerryescandon should chime in here, but I feel like they will always be a `fieldset`, because you're saving the list wrapper's form as a whole in order to save their order.